test/AutoDiff/transpose_attr_type_checking.swift

```diff
@@ -524,32 +522,64 @@ extension Struct {
 
 extension Struct where T: Differentiable & AdditiveArithmetic {
   @transpose(of: subscript, wrt: self)
-  func vjpSubscript() -> Self {
-    self
+  static func vjpSubscript(t: Struct) -> Struct {
+    t
   }
 
   @transpose(of: subscript(float:), wrt: self)
-  func vjpSubscriptLabelled(_ float: Float) -> Self {
-    self
+  static func vjpSubscriptLabelled(float: Float, t: Struct) -> Struct {
+    t
   }
 
   @transpose(of: subscript(_:), wrt: self)
-  func vjpSubscriptGeneric<U: Differentiable>(x: U) -> Self {
-    self
+  static func vjpSubscriptGeneric<U: Differentiable>(x: U, t: Struct) -> Struct {
+    t
   }
 }
 
 // Check that `@transpose` attribute rejects stored property original declarations.
 
-struct StoredProperty: Differentiable {
+struct StoredProperty: Differentiable & AdditiveArithmetic {
   var stored: Float
+  typealias TangentVector = StoredProperty
+  static var zero: StoredProperty { StoredProperty(stored: 0) }
+  static func + (_: StoredProperty, _: StoredProperty) -> StoredProperty {
+    StoredProperty(stored: 0)
+  }
+  static func - (_: StoredProperty, _: StoredProperty) -> StoredProperty {
+    StoredProperty(stored: 0)
+  }
 
   // Note: `@transpose` support for instance members is currently too limited
   // to properly register a transpose for a non-`Self`-typed member.
 
   // expected-error @+1 {{could not find function 'stored' with expected type '(StoredProperty) -> () -> StoredProperty'}}
   @transpose(of: stored, wrt: self)
-  func vjpStored() -> Self {
+  static func vjpStored(v: Self) -> Self {
     fatalError()
   }
 }
+
+// Check that the self type of the method and the result type are the same when 
```

Comment:
Please take care to avoid trailing whitespace if possible! It's not highlighted obviously on GitHub, but it is clear in `git diff`.
```suggestion
// Check that the self type of the method and the result type are the same when
```
I'll fix this and other occurrences now.